File: C# Net/VIQA/HtmlElements/Interfaces/ISelector.cs
using System.Collections.Generic;
using VIQA.HtmlElements.BaseClasses;

namespace VIQA.HtmlElements.Interfaces
{
    public interface ISelector<T> : IHaveValue, IVIList<T>
    {
        List<string> GetListOfValues();
        void Select(string valueName);
        bool IsSelected(string valueName);
    }

    public interface ISelector : ISelector<SelectItem> { }
}

File: Java/src/main/java/ru/viqa/ui_testing/elements/interfaces/ISelector.java
package ru.viqa.ui_testing.elements.interfaces;

import java.util.List;

/**
 * Created by roman.i on 29.09.2014.
 */
public interface ISelector<T extends Enum> extends IHaveValue {
    void select(String valueName) throws Exception;
    void select(T valueName) throws Exception;
    String getText(String valueName) throws Exception;
    List<String> getAllLabels() throws Exception;
    String isSelected() throws Exception;
}
